crates/rethnet_eth/src/remote/methods.rs

```diff
@@ -364,4 +441,98 @@ mod tests {
     fn test_zero_x_prefixed_bytes_deserialization_with_0_prefix_but_no_x() {
         serde_json::from_str::<ZeroXPrefixedBytes>("\"0deadbeef\"").unwrap();
     }
+
+    #[test]
+    fn test_infinite_percentile() {
+        assert!(Percentile::try_from(f64::INFINITY).is_err());
+    }
+
+    #[test]
+    fn test_nan_percentile() {
+        assert!(Percentile::try_from(f64::INFINITY).is_err());
+    }
+
+    #[test]
+    fn test_negative_percentile() {
+        assert!(Percentile::try_from(-1.).is_err());
+    }
+
+    #[test]
+    fn test_too_large_percentile() {
+        assert!(Percentile::try_from(1000.).is_err());
+    }
+
+    #[test]
+    fn test_zero_percentile() {
+        assert!(Percentile::try_from(0.).is_ok());
+    }
+
+    #[test]
+    fn test_hundred_percentile() {
+        assert!(Percentile::try_from(100.).is_ok());
+    }
+
+    #[test]
+    fn test_percentile_serialization() {
```

Comment:
Couldn't you combine the two tests into a round-trip test where you serialize an input, deserialize its output, and compare it to the source input?